cirq/__init__.py

```diff
@@ -86,10 +86,12 @@
 )
 
 from cirq.experiments import (
-    linear_xeb_fidelity,
+    linear_xeb_fidelity_estimator,
+    log_xeb_fidelity_estimator,
     generate_boixo_2018_supremacy_circuits_v2,
     generate_boixo_2018_supremacy_circuits_v2_bristlecone,
     generate_boixo_2018_supremacy_circuits_v2_grid,
+    xeb_fidelity,
```

Comment:
Just have `linear_xeb_fidelity` and `log_xeb_fidelity`. It's not necessary to parameterize this method so flexibly.

Reply:
XEB allows you to use any of a whole range of fidelity estimators of which log and linear are just two examples. The estimators differ in important ways such as in the assumptions they make about the output distribution(*) or in the conditions under which their estimate has good quality(**). Also, in some cases, it is interesting to investigate/plot results obtained from more than one estimator. The benefits of being able to replace the estimator while re-using the rest of the code is one of the reasons that motivated #2113.

BTW: Two additional estimators I plan to add:
1. HOG score based fidelity estimator (user requested ability to compare log/linear XEB vs HOG).
2. Cross-entropy ratio since it does not require the assumption that output distribution is PT (and hence may be used on shallow circuits).

----
(*) A common assumption is that the output distribution has Porter-Thomas shape which allows analytical derivation of the simple formulas in the code.
(**) For example, linear estimator is good in low-fidelity regime (e.g. large, deep circuits), but log estimator is better in high-fidelity regime (e.g. benchmarking of a 2-qubit gate).

Reply:
Do you believe that user code will also parameterize over the method in this way? I suspect users will always just be calling a specific one.

Basically, this is setting off my boilerplate alarms. The user wants the linear xeb fidelity, and in order to get that they need to provide the linear estimator to the xeb fidelity method. Why not just ask directly for what they want?

It's okay if we also have an abstract method behind the scenes, or exposed through some more obviously-for-when-generalizing name like "generalized_xeb_fidelity", but I do think we should directly expose `linear_xeb` and `log_xeb` methods.

Reply:
It makes sense to me to have functions dedicated to the common case and a general one to address the more nuanced cases requiring extra flexibility.

I kept the name `xeb_fidelity` for the general method and updated the docstring to point out the generality of this function (i.e. that it takes a user-supplied estimator). I think the name `generalized_xeb_fidelity` would suggest the overall methodology of XEB is somehow generalized here which is not the case. Also, in literature all these variants are referred to simply as "XEB" and when a specific case is meant we use terms like "linear XEB" or "log XEB". This agrees well with our present function naming.